preinstall/preinstall.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func deleteExistingObjects() {
+	csrCmd := exec.Command("kubectl", "get", "csr", csrName)
+	csrCmd.Stderr = os.Stderr
+	_, err := csrCmd.Output()
+	fmt.Println(err)
+	if err == nil {
+		deleteCSRCmd := exec.Command("kubectl", "delete", "csr", csrName)
+		RunCommand(deleteCSRCmd)
+	}
+	secretCmd := exec.Command("kubectl", "get", "secret", tlsSecretName)
+	secretCmd.Stderr = os.Stderr
+	_, err = secretCmd.Output()
+	fmt.Println(err)
+	if err == nil {
+		deleteSecretCmd := exec.Command("kubectl", "delete", "secret", tlsSecretName)
+		RunCommand(deleteSecretCmd)
+	}
+}
+
+func createCertificates() {
+	cert := `{
+"hosts": [
+    "kritis-validation-hook",
+    "kritis-validation-hook.kube-system",
+    "kritis-validation-hook.%s",
+    "kritis-validation-hook.%s.svc"
+],
+"key": {
+	"algo": "ecdsa",
+	"size": 256
+}
+}`
+	cert = fmt.Sprintf(cert, namespace, namespace)
+	certCmd := exec.Command("cfssl", "genkey", "-")
+	certCmd.Stdin = bytes.NewReader([]byte(cert))
+	output := RunCommand(certCmd)
+
+	serverCmd := exec.Command("cfssljson", "-bare", "server")
+	serverCmd.Stdin = bytes.NewReader(output)
+	RunCommand(serverCmd)
+}
+
+func createCertificateSigningRequest() {
+	certificate := retrieveRequestCertificate()
+	csr := `apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+    name: %s
```

Comment:
i think namespace shd go over here.

Reply:
The CertificateSigningRequests are global rather than namespaced